packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -118,22 +122,27 @@ export class PendingStateManager implements IDisposable {
 		return this.pendingMessagesCount !== 0;
 	}
 
-	public getLocalState(): IPendingLocalState | undefined {
+	public getLocalState(snapshotSequenceNumber?: number): IPendingLocalState {
 		assert(
 			this.initialMessages.isEmpty(),
 			0x2e9 /* "Must call getLocalState() after applying initial states" */,
 		);
-		if (!this.pendingMessages.isEmpty()) {
-			return {
-				pendingStates: [...this.savedOps, ...this.pendingMessages.toArray()].map(
-					(message) => {
-						// delete localOpMetadata since it may not be serializable
-						// and will be regenerated by applyStashedOp()
-						return { ...message, localOpMetadata: undefined };
-					},
-				),
-			};
-		}
+		const newSavedOps = [...this.savedOps].filter(
+			(message) => message.referenceSequenceNumber >= (snapshotSequenceNumber ?? 0),
```

Comment:
Is this correct?
If I got it right, the contract - we will load from snapshot with refSeq# === snapshotSequenceNumber.
If that's true, consider an op with
- refSeq# = 5
- seq# = 15
- and snapshotSequenceNumber  = 10.

We will process this op, and it will have same clientId as Container's clientID, so it will be considered "local" and pendingStateManger will bark that it can't find it, as it was excluded.

This test should be based on sequences numbers, not reference Sequence numbers!